File: starterAIs/SS_Starter.java
import java.util.*;

// Define the data structures as records
record Vector(int x, int y) {}

record FishDetail(int color, int type) {}

record Fish(int fishId, Vector pos, Vector speed, FishDetail detail) {}

record Drone(int droneId, Vector pos, boolean dead, int battery, List<Integer> scans) {}

record RadarBlip(int fishId, String dir) {}

class Player {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        Map<Integer, FishDetail> fishDetails = new HashMap<>();

        int fishCount = in.nextInt();
        for (int i = 0; i < fishCount; i++) {
            int fishId = in.nextInt();
            int color = in.nextInt();
            int type = in.nextInt();
            fishDetails.put(fishId, new FishDetail(color, type));
        }

        // game loop
        while (true) {
            List<Integer> myScans = new ArrayList<>();
            List<Integer> foeScans = new ArrayList<>();
            Map<Integer, Drone> droneById = new HashMap<>();
            List<Drone> myDrones = new ArrayList<>();
            List<Drone> foeDrones = new ArrayList<>();
            List<Fish> visibleFishes = new ArrayList<>();
            Map<Integer, List<RadarBlip>> myRadarBlips = new HashMap<>();

            int myScore = in.nextInt();
            int foeScore = in.nextInt();

            int myScanCount = in.nextInt();
            for (int i = 0; i < myScanCount; i++) {
                int fishId = in.nextInt();
                myScans.add(fishId);
            }

            int foeScanCount = in.nextInt();
            for (int i = 0; i < foeScanCount; i++) {
                int fishId = in.nextInt();
                foeScans.add(fishId);
            }

            

            int myDroneCount = in.nextInt();
            for (int i = 0; i < myDroneCount; i++) {
                int droneId = in.nextInt();
                int droneX = in.nextInt();
                int droneY = in.nextInt();
                boolean dead = in.nextInt() == 1;
                int battery = in.nextInt();
                Vector pos = new Vector(droneX, droneY);
                Drone drone = new Drone(droneId, pos, dead, battery, new ArrayList<>());
                droneById.put(droneId, drone);
                myDrones.add(drone);
                myRadarBlips.put(droneId, new ArrayList<>());
            }

            int foeDroneCount = in.nextInt();
            for (int i = 0; i < foeDroneCount; i++) {
                int droneId = in.nextInt();
                int droneX = in.nextInt();
                int droneY = in.nextInt();
                boolean dead = in.nextInt() == 1;
                int battery = in.nextInt();
                Vector pos = new Vector(droneX, droneY);
                Drone drone = new Drone(droneId, pos, dead, battery, new ArrayList<>());
                droneById.put(droneId, drone);
                foeDrones.add(drone);
            }

            int droneScanCount = in.nextInt();
            for (int i = 0; i < droneScanCount; i++) {
                int droneId = in.nextInt();
                int fishId = in.nextInt();
                droneById.get(droneId).scans().add(fishId);
            }

            int visibleFishCount = in.nextInt();
            for (int i = 0; i < visibleFishCount; i++) {
                int fishId = in.nextInt();
                int fishX = in.nextInt();
                int fishY = in.nextInt();
                int fishVx = in.nextInt();
                int fishVy = in.nextInt();
                Vector pos = new Vector(fishX, fishY);
                Vector speed = new Vector(fishVx, fishVy);
                FishDetail detail = fishDetails.get(fishId);
                visibleFishes.add(new Fish(fishId, pos, speed, detail));
            }

            int myRadarBlipCount = in.nextInt();
            for (int i = 0; i < myRadarBlipCount; i++) {
                int droneId = in.nextInt();
                int fishId = in.nextInt();
                String radar = in.next();
                myRadarBlips.get(droneId).add(new RadarBlip(fishId, radar));
            }

            for (Drone drone : myDrones) {
                int x = drone.pos().x();
                int y = drone.pos().y();
                // TODO: Implement logic on where to move here
                int targetX = 5000;
                int targetY = 5000;
                int light = 1;

                System.out.println(String.format("MOVE %d %d %d", targetX, targetY, light));
            }
        }
    }
}

File: starterAIs/SS_Starter.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

// Define the data structures as records
record Vector(int x, int y) { }

record FishDetail(int color, int type) { }

record Fish(int fishId, Vector pos, Vector speed, FishDetail detail) { }

record Drone(int droneId, Vector pos, bool dead, int battery, List<int> scans) { }

record RadarBlip(int fishId, String dir) { }

/**
 * Score points by scanning valuable fish faster than your opponent.
 **/
class Player
{
    static void Main(string[] args)
    {
        string[] inputs;

        Dictionary<int, FishDetail> fishDetails = new Dictionary<int, FishDetail>();

        int fishCount = int.Parse(Console.ReadLine());
        for (int i = 0; i < fishCount; i++)
        {
            inputs = Console.ReadLine().Split(' ');
            int fishId = int.Parse(inputs[0]);
            int color = int.Parse(inputs[1]);
            int type = int.Parse(inputs[2]);
            fishDetails.Add(fishId, new FishDetail(color, type));
        }

        // game loop
        while (true)
        {
            List<int> myScans = new List<int>();
            List<int> foeScans = new List<int>();
            Dictionary<int, Drone> droneById = new Dictionary<int, Drone>();
            List<Drone> myDrones = new List<Drone>();
            List<Drone> foeDrones = new List<Drone>();
            List<Fish> visibleFishes = new List<Fish>();
            Dictionary<int, List<RadarBlip>> myRadarBlips = new Dictionary<int, List<RadarBlip>>();

            int myScore = int.Parse(Console.ReadLine());
            int foeScore = int.Parse(Console.ReadLine());
            int myScanCount = int.Parse(Console.ReadLine());
            for (int i = 0; i < myScanCount; i++)
            {
                int fishId = int.Parse(Console.ReadLine());
                myScans.Add(fishId);
            }
            int foeScanCount = int.Parse(Console.ReadLine());
            for (int i = 0; i < foeScanCount; i++)
            {
                int fishId = int.Parse(Console.ReadLine());
                foeScans.Add(fishId);
            }
            int myDroneCount = int.Parse(Console.ReadLine());
            for (int i = 0; i < myDroneCount; i++)
            {
                inputs = Console.ReadLine().Split(' ');
                int droneId = int.Parse(inputs[0]);
                int droneX = int.Parse(inputs[1]);
                int droneY = int.Parse(inputs[2]);
                bool dead = int.Parse(inputs[3]) == 1;
                int battery = int.Parse(inputs[4]);
                Vector pos = new Vector(droneX, droneY);
                Drone drone = new Drone(droneId, pos, dead, battery, new List<int>());
                droneById.Add(droneId, drone);
                myDrones.Add(drone);
                myRadarBlips.Add(droneId, new List<RadarBlip>());
            }
            int foeDroneCount = int.Parse(Console.ReadLine());
            for (int i = 0; i < foeDroneCount; i++)
            {
                inputs = Console.ReadLine().Split(' ');
                int droneId = int.Parse(inputs[0]);
                int droneX = int.Parse(inputs[1]);
                int droneY = int.Parse(inputs[2]);
                bool dead = int.Parse(inputs[3])==1;
                int battery = int.Parse(inputs[4]);
                Vector pos = new Vector(droneX, droneY);
                Drone drone = new Drone(droneId, pos, dead, battery, new List<int>());
                droneById.Add(droneId, drone);
                foeDrones.Add(drone);
            }
            int droneScanCount = int.Parse(Console.ReadLine());
            for (int i = 0; i < droneScanCount; i++)
            {
                inputs = Console.ReadLine().Split(' ');
                int droneId = int.Parse(inputs[0]);
                int fishId = int.Parse(inputs[1]);
                droneById[droneId].scans.Add(fishId);
            }
            int visibleCreatureCount = int.Parse(Console.ReadLine());
            for (int i = 0; i < visibleCreatureCount; i++)
            {
                inputs = Console.ReadLine().Split(' ');
                int fishId = int.Parse(inputs[0]);
                int fishX = int.Parse(inputs[1]);
                int fishY = int.Parse(inputs[2]);
                int fishVx = int.Parse(inputs[3]);
                int fishVy = int.Parse(inputs[4]);
                Vector pos = new Vector(fishX, fishY);
                Vector speed = new Vector(fishVx, fishVy);
                FishDetail detail = fishDetails[fishId];
                visibleFishes.Add(new Fish(fishId, pos, speed, detail));
            }
            int radarBlipCount = int.Parse(Console.ReadLine());
            for (int i = 0; i < radarBlipCount; i++)
            {
                inputs = Console.ReadLine().Split(' ');
                int droneId = int.Parse(inputs[0]);
                int fishId = int.Parse(inputs[1]);
                string radar = inputs[2];
                myRadarBlips[droneId].Add(new RadarBlip(fishId, radar));
            }
            foreach (Drone drone in myDrones)
            {
                int x = drone.pos.x;
                int y = drone.pos.y;
                // TODO: Implement logic on where to move here
                int targetX = 5000;
                int targetY = 5000;
                int light = 1;

                Console.WriteLine(String.Format("MOVE {0} {1} {2}", targetX, targetY, light));
            }
        }
    }
}